### PR TITLE
Cal/vnext

### PR DESCRIPTION
**Pull request by McCallisterRomer**

### CODE REVIEW

File: nc/source/ui/Editor.cpp [hunk 1]
```diff
@@ -0,0 +1,151 @@
+#ifdef NC_EDITOR_ENABLED
```
Comment (McCallisterRomer) [inline nc/source/ui/Editor.cpp:1]: This is just EditorManager renamed

File: project/Components/CamController.cpp [hunk 2]
```diff
@@ -7,11 +7,12 @@ using namespace nc;
 namespace
 {
     const float CAM_ZOOM_SPEED = 2.5f;
-    const float CAM_PAN_SPEED = 25.0f;
-    const float CAM_ROTATE_SPEED = -1.2f;
+    const float CAM_PAN_SPEED = 70.0f;
+    const unsigned EDGE_PAN_WIDTH = 180;
```
Comment (nightcrawlerstudios) [inline project/Components/CamController.cpp:11]: Edge panning! <3

File: project/ui/HUD.cpp [hunk 3]
```diff
@@ -0,0 +1,102 @@
+#include "HUD.h"
+#include "imgui/imgui.h"
+#include "graphics/d3dresource/GraphicsResource.h"
+#include "LogUIElement.h"
+#include "TurnPhaseUIElement.h"
+
+namespace
+{
+    constexpr auto HUD_WINDOW_FLAGS = ImGuiWindowFlags_MenuBar | ImGuiWindowFlags_NoCollapse | ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoResize;
```
Comment (nightcrawlerstudios) [inline project/ui/HUD.cpp:9]: This line is getting kind of wide. I think vertically aligning theses would look nicer.
ImGuiWindowFlags_MenuBar |
ImGuiWindowFlags_NoCollapse | 
ImGuiWindowFlags_NoTitleBar | 
ImGuiWindowFlags_NoResize;

File: project/ui/HUD.cpp [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    constexpr unsigned RESOURCE_ICON_SIZE = 16;
+}
+
+namespace project::ui
+{
+    HUD::HUD()
+        : m_logUIElement {true},
+          m_turnPhaseUIElement {true},
+          m_texture { std::make_unique<nc::graphics::d3dresource::Texture>("project/Textures/icon.bmp", 0) }
```
Comment (nightcrawlerstudios) [inline project/ui/HUD.cpp:18]: It'd be nice to expose the texture path as a param.

**Review by nightcrawlerstudios — APPROVED** [review]

**Pull request merged**